Error in service log reader closing

### Stacktrace

```
[17.04 01:37:23.819] ERROR: Unable to read out/err stream of service lobby-1:0c9a8833-9156-4652-9070-85676387b63b
java.lang.NullPointerException: Cannot invoke "java.io.Reader.close()" because "this.reader" is null
    syncat eu.cloudnetservice.node.impl.service.defaults.log.NonBlockingLineReader.close(NonBlockingLineReader.java:185)
        at eu.cloudnetservice.node.impl.service.defaults.log.ProcessServiceLogCache.stop(ProcessServiceLogCache.java:63)
        at eu.cloudnetservice.node.impl.service.defaults.log.ProcessServiceLogCache.readProcessOutputContent(ProcessServiceLogCache.java:99)
        at eu.cloudnetservice.node.impl.service.defaults.log.ProcessServiceLogReadScheduler$ProcessServiceLogReadTask.run(ProcessServiceLogReadScheduler.java:84)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:545)
        at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:369)
        at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:310)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
        at java.base/java.lang.Thread.run(Thread.java:1474)
```


### Actions to reproduce

Can't reproduce, seems like a race condition. 
Happened exactly as the service was done stopping, might be coincidence but I don't think so.
My guess is the console reader thread detected that the service is dead. Then it tried to stop/close the cache reader and the JVMService
stopService call raced, since they both call `ProcessServiceLogCache#stop` and run on different threads

### CloudNet version

[17.04 01:43:05.605] INFO :
[17.04 01:43:05.607] INFO : CloudNet Blizzard 4.0.0-RC17-SNAPSHOT 4b3e489e
[17.04 01:43:05.608] INFO : Discord: <https://discord.cloudnetservice.eu/>
[17.04 01:43:05.609] INFO :
[17.04 01:43:05.609] INFO : ClusterId: cff1795a-****-4230-****-59d7342463c3
[17.04 01:43:05.610] INFO : NodeId: Node-1
[17.04 01:43:05.611] INFO : Head-NodeId: Node-1
[17.04 01:43:05.612] INFO : CPU usage: (P/S) 5.04/45.34/100%
[17.04 01:43:05.613] INFO : Node services memory allocation (U/R/M): 2524/2524/10000 MB
[17.04 01:43:05.614] INFO : Threads: 49
[17.04 01:43:05.614] INFO : Heap usage: 59/256MB
[17.04 01:43:05.615] INFO : JVM: Eclipse Adoptium 25 (OpenJDK 64-Bit Server VM 25.0.2+10-LTS)
[17.04 01:43:05.616] INFO : Update Repo: CloudNetService/launchermeta, Update Branch: beta (development mode)
[17.04 01:43:05.616] INFO :

### Other

_No response_

### Issue uniqueness

- [x] Yes, this issue is unique. There are no similar issues.

Happened again today, on c019e30f760d97b8ff78eb3c021effb1025a7f4e with a different error, again as the service was done stopping.
Might be the other side of the race condition
```
[29.07 15:57:38.367] ERROR: Exception during command execution
org.incendo.cloud.exception.CommandExecutionException: java.lang.NullPointerException: Cannot invoke "java.io.Reader.close()" because "this.reader" is null
        at org.incendo.cloud.annotations.MethodCommandExecutionHandler.executeFuture(MethodCommandExecutionHandler.java:100)
        at org.incendo.cloud.execution.ExecutionCoordinatorImpl.lambda$coordinateExecution$4(ExecutionCoordinatorImpl.java:121)
        at java.base/java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1171)
        at java.base/java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:503)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
        at java.base/java.lang.Thread.run(Thread.java:1474)
Caused by: java.lang.NullPointerException: Cannot invoke "java.io.Reader.close()" because "this.reader" is null
        at eu.cloudnetservice.node.impl.service.defaults.log.NonBlockingLineReader.close(NonBlockingLineReader.java:185)
        at eu.cloudnetservice.node.impl.service.defaults.log.ProcessServiceLogCache.stop(ProcessServiceLogCache.java:63)
        at eu.cloudnetservice.node.impl.service.defaults.JVMService.stopProcess(JVMService.java:220)
        at eu.cloudnetservice.node.impl.service.defaults.AbstractService.doDelete(AbstractService.java:545)
        at eu.cloudnetservice.node.impl.service.defaults.AbstractService.updateLifecycle(AbstractService.java:386)
        at eu.cloudnetservice.node.impl.service.defaults.AbstractService.updateLifecycle(AbstractService.java:252)
        at eu.cloudnetservice.driver.provider.SpecificCloudServiceProvider.stop(SpecificCloudServiceProvider.java:154)
        at eu.cloudnetservice.node.impl.command.sub.ServiceCommand.stopServices(ServiceCommand.java:194)
        at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:735)
        at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:760)
        at org.incendo.cloud.annotations.MethodCommandExecutionHandler.executeFuture(MethodCommandExecutionHandler.java:88)
        ... 6 common frames omitted
```

--- a/node/impl/src/main/java/eu/cloudnetservice/node/impl/service/defaults/log/NonBlockingLineReader.java
+++ b/node/impl/src/main/java/eu/cloudnetservice/node/impl/service/defaults/log/NonBlockingLineReader.java
@@ -182,7 +182,10 @@ final class NonBlockingLineReader implements Closeable {
    */
   @Override
   public void close() throws IOException {
-    this.reader.close();
-    this.reader = null;
+    var reader = this.reader;
+    if (reader != null) {
+      this.reader = null;
+      reader.close();
+    }
   }
 }

--- a/node/impl/src/test/java/eu/cloudnetservice/node/impl/service/defaults/log/NonBlockingLineReaderTest.java
+++ b/node/impl/src/test/java/eu/cloudnetservice/node/impl/service/defaults/log/NonBlockingLineReaderTest.java
@@ -48,6 +48,15 @@ public class NonBlockingLineReaderTest {
   }
 
   @Test
+  void testClosingReaderIsIdempotent() {
+    var source = new StringReader("Hello\nWorld");
+    var reader = new NonBlockingLineReader(source);
+
+    Assertions.assertDoesNotThrow(reader::close);
+    Assertions.assertDoesNotThrow(reader::close);
+  }
+
+  @Test
   void testMultipleNewlinesAreParsedCorrectly() throws IOException {
     var source = new StringReader("Hello\n\nWorld\n");
     var reader = new NonBlockingLineReader(source);

